sdk/parents/ci.yml

```diff
@@ -31,11 +31,18 @@ pr:
   paths:
     include:
       - sdk/parents/
-
+  
 stages:
-  - stage: Placeholder
-    jobs:
-      - job: Placeholder
-        steps:
-          - script: |
-              echo Hello, world!
+  - template: ../../eng/pipelines/templates/stages/archetype-sdk-pom-only.yml
```

Comment:
I'm under the impression that this is mainly for management track right now so why isn't that artifact listed here? In theory, we're done releasing client and data parent poms and have been for months.

Reply:
There is a challenge with management plane right now, specifically it is pointing to an old parent POM which has a distribution management section. As part of this PR I remove the distribution management section from ```azure-sdk-parent``` since it gets in the road (and is kinda pointless anyway in our context).

I think there are some changes coming that will result in the parent POMs being updated (I think @alzimmermsft  was working on something recently).

Reply:
... oh and this same pipeline will also be used by BOM deployments.

Reply:
To be 100% clear: the client and data parent poms should really not need to be released ever again. The only parent pom that we know we're guaranteed to have to release at some point and time in the future is azure-arm-parent which management's parent. That should definitely be one of the artifacts listed below.